packages/mui-system/src/colorManipulator/colorManipulator.js

```diff
@@ -36,7 +36,16 @@ export function hexToRgb(color) {
   return colors
     ? `rgb${colors.length === 4 ? 'a' : ''}(${colors
         .map((n, index) => {
-          return index < 3 ? parseInt(n, 16) : Math.round((parseInt(n, 16) / 255) * 1000) / 1000;
+          if (index < 3) {
+            return parseInt(n, 16);
+          }
+          const channel = Math.round((parseInt(n, 16) / 255) * 1000) / 1000;
+          if (Number.isNaN(channel)) {
```

Comment:
Maybe we should test the input beforehand? It's not clear that we can conclude that NaN <> leading space.

```suggestion
          if (n === '') {
```

Reply:
Also, I think the whole point of the initial suggestion was to have:

```suggestion
          if (process.env.NODE_ENV !== 'production') {
            if (Number.isNaN(channel)) {
```

meaning the DX win without the UX bloat.

Reply:
Updated